include/graph_configuration.h

```diff
@@ -0,0 +1,93 @@
+
+#include <iostream>
+static constexpr char config_loc[] = "streaming.conf";
+
+// TODO: Replace this with an enum defined by GutterTree repo
+enum GutterSystem {
+  GUTTERTREE,
+  STANDALONE,
+  CACHETREE
+};
+
+// Graph parameters
+class GraphConfiguration {
+public:
+  // which guttering system to use for buffering updates
+  GutterSystem gutter_sys = STANDALONE;
+
+  // Where to place on-disk datastructures
+  std::string disk_dir = ".";
+
+  // Backup supernodes in memory or on disk when performing queries
+  bool backup_in_mem = true;
+
+  // The number of graph workers
+  size_t num_groups = 1;
+
+  // How many OMP threads each graph worker uses
+  size_t group_size = 1;
+
+  void read_configuration() {
+    std::ifstream config_file(config_loc);
+    std::string line;
+    if (config_file.is_open()) {
+      while(getline(config_file, line)) {
+        if (line[0] == '#' || line[0] == '\n') continue;
+        if(line.substr(0, line.find('=')) == "buffering_system") {
+          std::string buf_str = line.substr(line.find('=') + 1);
+          if (buf_str == "tree") gutter_sys = GUTTERTREE;
+          else if (buf_str == "cachetree") gutter_sys = CACHETREE;
+          else if (buf_str != "standalone") {
```

Comment:
While this is correct, changing the default makes it incorrect.

Reply:
Oh, do you mean that changing the default gutter_sys would break this logic?

Reply:
Yes, exactly that.